recipes/MemoizeFibonacci/src/Main.purs

```diff
@@ -0,0 +1,53 @@
+module MemoizeFibonacci.Main where
+
+import Prelude
+import Data.Function.Memoize (memoize)
+import Data.Interpolate (i)
+import Debug.Trace (spy)
+import Effect (Effect)
+import Effect.Class.Console (log)
+
+-- Basic fibonacci implementation
+fib :: Int -> Int
+fib 0 = 0
+fib 1 = 1
+fib n = fib (n - 2) + fib (n - 1)
+
+-- Same as `fib`, but with `spy` statements added to reveal duplicated work
+fibSlow :: Int -> Int
+fibSlow 0 = spy "fibSlow 0" 0
+fibSlow 1 = spy "fibSlow 1" 1
+fibSlow n =
+  spy ("fibSlow " <> show n)
+    $ fibSlow (n - 2)
+    + fibSlow (n - 1)
+
+-- Fast memoized version of fibonacci
+fibFast :: Int -> Int
+fibFast 0 = spy "fibFast 0" 0
+fibFast 1 = spy "fibFast 1" 1
+fibFast n =
+  spy ("fibFast " <> show n)
+    $ fibMemo (n - 2)
+    + fibMemo (n - 1)
+
+fibMemo :: Int -> Int
+fibMemo = memoize \n -> fibFast n
+
+{-
+Note that the following adjustments to `fibMemo` eliminate the
+performance benefits of `memoize`, even though they may still
+type-check. Feel free to experiment with these yourself:
+
+* Rewrite in non-thunked or point-free form:
+  * `fibMemo n = memoize fibFast n`
+  * `fibMemo = memoize fibFast`
+* Move to a `let` or `where` clause inside of `fibFast`.
+-}
```

Comment:
If I'm understanding you correctly, adjusting `fibMemo` in the above 3 ways will remove the performance benefit of memoization. 
Could we convert those examples to code, add them as statements to `main` but leave those statements commented out with a header above explaining why?
For example:
```purescript
main :: Effect Unit
main = do
  log $ i "basic fib result" $ fib 7
  ...
  log $ i "fibSlow result: " $ fibSlow 7

  -- These examples type check and seem to implement
  -- to incur the performance benefits of memoization. 
  -- However, they do not. Uncomment them to see for yourself.
  -- log $ i "fibBroken1 result: " $ fibBroken1 7
  -- ..
```
I think I'd also prefer that `main` be immediately below the import statements, so that one can see what will run and then see how they are defined.

Thoughts?
```

Reply:
ping @milesfrain

Reply:
Yeah. These are good suggestions. I'll circle back to this task in about a week. Trying to wrap-up the try-purescript enhancements first.

Reply:
In that case, I'll push changes to this PR and merge. If you have feedback on what I've done, feel free to open an issue.

Stay focused on the try-purescript stuff as that's more important.